### PR TITLE
Nav and Tile: Add ability for screen reader to read selected state

### PR DESCRIPTION
**Pull request by inateeg**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Include a change request file using `$ npm run change`

#### Description of changes

Accessibility fixes:  Screen reader need to read the Nav link selected state if selected;  Tile also missing selected state.

#### Focus areas to test

Screen reader

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/9777)

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/Nav/Nav.base.tsx [hunk 1]
```diff
@@ -87,7 +88,8 @@ export class NavBase extends React.Component<INavProps, INavState> implements IN
 
     // Prevent hijacking of the parent window if link.target is defined
     const rel = link.url && link.target && !isRelativeUrl(link.url) ? 'noopener noreferrer' : undefined;
-
+    const selectStateAria = isSelectedLink ? 'Selected' : '';
```
Comment (inateeg) [inline packages/office-ui-fabric-react/src/components/Nav/Nav.base.tsx]: I am looking at how to get localized string here.

**Comment by inateeg** [comment]
since resource string need to be passed in.  explore possible way for caller update link aria label for selected state info

**Comment by inateeg** [comment]
I added prop change to pass in the localized resource string.

File: packages/office-ui-fabric-react/src/components/Nav/Nav.base.tsx [hunk 2]
```diff
@@ -99,7 +104,7 @@ export class NavBase extends React.Component<INavProps, INavState> implements IN
         target={link.target}
         rel={rel}
         disabled={link.disabled}
-        aria-label={link.ariaLabel}
+        aria-label={`${ariaLabel} ${selectStateAria}`}
```
Comment (Vitalius1) [inline packages/office-ui-fabric-react/src/components/Nav/Nav.base.tsx]: You will be having at least a space character aria-label if both are not provided, an `ariaLabel` with a space at the end if the `selectAriaState` missing or a `selectAriaState` with a space at the beginning if `ariaLabel` missing. This is actually reflected in your snapshot updates. Perhaps a ternary would be helpful in here:
```suggestion
        aria-label={ariaLabel + selectStateAria ? ` ${selectStateAria}` : ‘’}
```

File: packages/office-ui-fabric-react/src/components/__snapshots__/Nav.Basic.Example.tsx.shot [hunk 3]
```diff
@@ -174,6 +174,7 @@ exports[`Component Examples renders Nav.Basic.Example.tsx correctly 1`] = `
                 </i>
               </button>
               <a
+                aria-label=" "
```
Comment (Vitalius1) [inline packages/office-ui-fabric-react/src/components/__snapshots__/Nav.Basic.Example.tsx.shot]: here the snapshots is updated with an empty space. So I assume the example is not setting any of aria labels.

Comment (inateeg) [reply packages/office-ui-fabric-react/src/components/__snapshots__/Nav.Basic.Example.tsx.shot]: thanks I should update the example here to pass in selectedAriaLabel

**Comment by khmakoto** [comment]
Hi @inateeg, seems like there are some merge conflicts. Could you update your PR to fix them? Thanks!

File: packages/office-ui-fabric-react/src/components/__snapshots__/Nav.Basic.Example.tsx.shot [hunk 4]
```diff
@@ -331,6 +332,7 @@ exports[`Component Examples renders Nav.Basic.Example.tsx correctly 1`] = `
                   name="Activity"
                 >
                   <a
+                    aria-label=""
```
Comment (Vitalius1) [inline packages/office-ui-fabric-react/src/components/__snapshots__/Nav.Basic.Example.tsx.shot]: Will an empty string for `aria-label` be processed correctly by AT?

Comment (inateeg) [reply packages/office-ui-fabric-react/src/components/__snapshots__/Nav.Basic.Example.tsx.shot]: should not have '' aria-label' fixed

File: packages/office-ui-fabric-react/src/components/Nav/Nav.base.tsx [hunk 5]
```diff
@@ -32,10 +32,13 @@ export class NavBase extends React.Component<INavProps, INavState> implements IN
   public static defaultProps: INavProps = {
     groups: null
   };
+  private _linkSelectedAriaLabel: string | undefined;
```
Comment (Vitalius1) [inline packages/office-ui-fabric-react/src/components/Nav/Nav.base.tsx]: I don’t think that an aria prop needs an instance property of its own. Enough to just add the prop to the props object destructuring on the line 80

File: packages/office-ui-fabric-react/src/components/Nav/Nav.base.tsx [hunk 6]
```diff
@@ -99,7 +103,15 @@ export class NavBase extends React.Component<INavProps, INavState> implements IN
         target={link.target}
         rel={rel}
         disabled={link.disabled}
-        aria-label={link.ariaLabel}
+        aria-label={
+          !!link.ariaLabel && !!selectStateAria
+            ? `${link.ariaLabel} ${selectStateAria}`
+            : !!selectStateAria
```
Comment (Vitalius1) [inline packages/office-ui-fabric-react/src/components/Nav/Nav.base.tsx]: The logic seems correct and in some degree able to follow.

**Pull request merged**